File: backend/app/models/habit.py
from pydantic import BaseModel
from typing import Optional
from datetime import date

class HabitBase(BaseModel):
    name: str
    emoji: str
    habit_type: str  # 'time', 'binary', 'description'
    color: Optional[str] = None
    active: bool = True
    order: int = 0
    is_personal: bool = False

class Habit(HabitBase):
    id: str
    current_streak: int = 0
    best_streak: int = 0
    completed_today: bool = False
    last_completion_date: Optional[date] = None

class HabitEntry(BaseModel):
    habit_id: str
    date: date
    value: Optional[str] = None
    completed: bool = False

File: frontend/src/types/habit.ts
export type HabitType = 'time' | 'binary' | 'description' | 'grade'

export interface Habit {
  id: string
  name: string
  emoji: string
  icon?: string
  habit_type: HabitType
  category?: string
  color?: string
  active: boolean
  order: number
  is_personal: boolean
  current_streak: number
  best_streak: number
  completed_today: boolean
  last_completion_date?: string
}

export interface HabitEntry {
  habit_id: string
  date: string
  value?: string
  completed: boolean
}

export interface Analytics {
  total_habits: number
  active_streaks: number
  longest_streak: number
  completed_today: number
  completion_rate: number
}

export interface CalendarDay {
  date: string
  completed_habits: number
  total_habits: number
  productivity_minutes: number
  perfect_day: boolean
  workout_grade?: string
}

export interface ExternalLink {
  name: string
  url: string
  icon: string
}